gobblin-runtime/src/main/java/org/apache/gobblin/runtime/spec_catalog/FlowCatalog.java

```diff
@@ -383,10 +387,17 @@ public Map<String, AddSpecResponse> put(Spec spec, boolean triggerListener) thro
         return responseMap;
       }
     }
-    AddSpecResponse<String> schedulerResponse = responseMap.getOrDefault(ServiceConfigKeys.GOBBLIN_SERVICE_JOB_SCHEDULER_LISTENER_CLASS, new AddSpecResponse<>(null));
+    AddSpecResponse<String> compileResponse;
+    if (isWarmStandbyEnabled) {
+      compileResponse = responseMap.getOrDefault(ServiceConfigKeys.GOBBLIN_ORCHESTRATOR_LISTENER_CLASS, new AddSpecResponse<>(null));
+      //todo: do we check quota here? or in compiler? Quota manager need dag to check quota which is not accessable from this class
```

Comment:
This compilation check is only to figure out if the flow should be persisted in the flow catalog or not, right? In that case, quota check is not required, correct?

Reply:
Quota check should only be enabled for adhoc flows or flows who are set to run immediately, otherwise it should be safe to store. We can't block users from adding flows to be scheduled in the future even if their current quota is overrun.

Reply:
When I say check quota, it's only for adhoc and run immediate flow. Since after this step, we put the change in mysql table and return accept to user which means we will run the flow anyway on leader host. So before we put it into mysql table, for achoc and run immediate flow, we need to check the quota first, either here or in compiler